### PR TITLE
Add test to notice upstream changes in group perferred versions

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/openshift/origin/pull/13653#discussion_r112762642.

### CODE REVIEW

**Comment by sttts** [comment]
[test]

File: pkg/cmd/server/kubernetes/master/master_config_test.go [hunk 1]
```diff
@@ -25,6 +26,31 @@ import (
 	configapi "github.com/openshift/origin/pkg/cmd/server/api"
 )
 
+var expectedGroupPreferredVersions []string = []string{
+	"apps/v1beta1,authentication.k8s.io/v1",
```
Comment (liggitt) [inline pkg/cmd/server/kubernetes/master/master_config_test.go]: one version per line for readability

Comment (liggitt) [inline pkg/cmd/server/kubernetes/master/master_config_test.go]: if the order isn't significant (I can't remember if it is), then sort both lists

**Comment by liggitt** [comment]
a couple nits, LGTM otherwise

File: pkg/cmd/server/kubernetes/master/master_config_test.go (hunk 1 above)
Comment (sttts) [inline pkg/cmd/server/kubernetes/master/master_config_test.go]: What we get back from AllPreferredGroupVersions is sorted already. Will add a comment above to keep it sorted.

Comment (sttts) [inline pkg/cmd/server/kubernetes/master/master_config_test.go]: done

**Comment by sttts** [comment]
[merge]

**Comment by sttts** [comment]
Flake https://github.com/openshift/origin/issues/12927

**Comment by sttts** [comment]
[test]

**Comment by sttts** [comment]
[merge]

**Comment by deads2k** [comment]
[merge]

**Comment by deads2k** [comment]
test fixes.  approved.

**Pull request merged**